File: dotnet/DesignPatterns/BehavioralDesignPatterns/ObserverDesignPattern/ObserverDesignPattern.Example2/ForecastDisplay.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ObserverDesignPattern.Example2;

public class ForecastDisplay : IObserver
{
    private float currentPressure = 1013.0f;
    private float lastPressure;
    private ISubject _weatherStation;

    public ForecastDisplay(ISubject weatherStation)
    {
        _weatherStation = weatherStation;
        _weatherStation.RegisterObserver(this);
    }

    public void Update(float temperature, float humidity, float pressure)
    {
        lastPressure = currentPressure;
        currentPressure = pressure;
        Display();
    }

    public void Display()
    {
        Console.Write("🔮 Weather Forecast: ");

        if (currentPressure > lastPressure)
        {
            Console.WriteLine("Improving!");
        }
        else if (currentPressure == lastPressure)
        {
            Console.WriteLine("No change.");
        }
        else if (currentPressure < lastPressure)
        {
            Console.WriteLine("Bad weather incoming.");
        }
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/ObserverDesignPattern/ObserverDesignPattern.Example2/src/ForecastDisplay.ts
import { IObserver } from './IObserver';
import { ISubject } from './ISubject';

export class ForecastDisplay implements IObserver {
  private _currentPressure: number = 1013.0;
  private _lastPressure: number = 1013.0;
  private readonly _weatherStation: ISubject;

  constructor(weatherStation: ISubject) {
    this._weatherStation = weatherStation;
    this._weatherStation.registerObserver(this);
  }

  update(_temperature: number, _humidity: number, pressure: number): void {
    this._lastPressure = this._currentPressure;
    this._currentPressure = pressure;
    this.display();
  }

  display(): void {
    process.stdout.write('🔮 Weather Forecast: ');
    if (this._currentPressure > this._lastPressure) {
      console.log('Improving!');
    } else if (this._currentPressure === this._lastPressure) {
      console.log('No change.');
    } else {
      console.log('Bad weather incoming.');
    }
  }
}
